test/unit/range/decorator/gap_decorator_anchor_set_test.cpp

```diff
@@ -0,0 +1,243 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include <seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp>
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/range/decorator/gap_decorator_anchor_set.hpp>
+#include <seqan3/std/view/subrange.hpp>
+#include <seqan3/std/view/view_all.hpp>
+#include <seqan3/std/ranges>
+
+#include "../../alignment/aligned_sequence_test_template.hpp"
+
+using namespace seqan3;
+
+template <typename inner_type_>
+class aligned_sequence<gap_decorator_anchor_set<inner_type_>> : public ::testing::Test
+{
+public:
+    // Initializer function is needed for the typed test because the gapped_decorator
+    // will be initialized differently than the naive vector<gapped<dna>>.
+    void initialize_typed_test_container(gap_decorator_anchor_set<inner_type_> & container, dna4_vector const & target)
+    {
+        container = target;
+    }
+};
+
+template <typename t>
+class gap_decorator_anchor_set_f : public ::testing::Test {};
+
+const std::vector<dna4> dummy_obj{}; // dummy lvalue for type declaration of views
+
+using decorator_t = gap_decorator_anchor_set<decltype(view::all(dummy_obj))>;
+using decorator_t2 = gap_decorator_anchor_set<
+                         decltype(view::subrange<decltype(dummy_obj.begin()),
+                                                 decltype(dummy_obj.begin())>{dummy_obj.begin(), dummy_obj.end()})>;
+
+using test_types = ::testing::Types<decorator_t, decorator_t2>;
+
+INSTANTIATE_TYPED_TEST_CASE_P(gap_decorator_anchor_set, aligned_sequence, test_types);
+
+TYPED_TEST_CASE(gap_decorator_anchor_set_f, test_types);
+
+// concept checks
+TYPED_TEST(gap_decorator_anchor_set_f, concept_checks)
+{
+    EXPECT_TRUE((std::ranges::RandomAccessRange<TypeParam>));
+    EXPECT_FALSE((std::ranges::enable_view<TypeParam>::value));
+    EXPECT_FALSE((ranges::enable_view<TypeParam>::value));
+    EXPECT_FALSE((std::ranges::View<TypeParam>));
+}
+
+TYPED_TEST(gap_decorator_anchor_set_f, construction_general)
+{
+    // default
+    {
+        [[maybe_unused]] TypeParam dec{};
+        [[maybe_unused]] TypeParam dec3{};
+        EXPECT_EQ(dec, dec3);
+    }
+
+    // copy
+    {
+        TypeParam dec{};
+        TypeParam dec2(dec);
+        EXPECT_EQ(dec, dec2);
+
+        TypeParam const dec_const{};
+        TypeParam const dec2_const(dec_const);
+        EXPECT_EQ(dec_const, dec2_const);
+    }
+
+    // copy assignment
```

Comment:
This is still copy constructor.

```
        TypeParam dec{};
        TypeParam dec2;
        dec2 = dec;
        EXPECT_EQ(dec, dec2);
```
would be assignment.

Then the const version will also not work (copy assignment should not work for a const type).

See https://en.cppreference.com/w/cpp/language/copy_constructor

Same for move assignment.